src/main/java/org/osiam/storage/helper/NumberPadder.java

```diff
@@ -56,7 +56,11 @@ public String pad(String value) {
             throw new IllegalArgumentException("The given value has more than " + (PAD_LENGTH - 1) + " digits.");
         }
 
-        integralPart = new BigInteger(integralPart).add(BIG_OFFSET).toString();
+        try {
+            integralPart = new BigInteger(integralPart).add(BIG_OFFSET).toString();
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException("Unable to parse a number from " + value);
```

Comment:
We could discuss logging this error as debug or something. I think this would be a different pull request.